client/packages/host/src/components/Sync/Sync.tsx

```diff
@@ -0,0 +1,113 @@
+import React, { PropsWithChildren, useState, useEffect } from 'react';
+
+import {
+  Grid,
+  LoadingButton,
+  RadioIcon,
+  Typography,
+  useFormatDateTime,
+  useTranslation,
+} from '@openmsupply-client/common';
+import { useHost } from '../../api/hooks';
+import { SyncProgress } from '../SyncProgress';
+
+const STATUS_POLLING_INTERVAL = 1000;
+
+const useSync = () => {
+  const { localisedDate, localisedTime } = useFormatDateTime();
+  // Polling whenever Sync page is opened
+  const { data } = useHost.utils.syncInfo(STATUS_POLLING_INTERVAL);
+  const { mutateAsync: manualSync } = useHost.sync.manualSync();
+
+  // Derived state
+  const [latestSyncDate, setLatestSyncDate] = useState<null | string>(null);
+  // true by default to wait for first syncStatus api result
+  const [isLoading, setIsLoading] = useState(true);
```

Comment:
if you use `loading` from `syncInfo` then isn't this always `loading || syncStatus === 'isSyncing'`?

Reply:
hmm I thought loading from useQuery is only when api call happens, not during polling delay, so button might not go into loading until next poll happens.

i.e. polling interval = 1second
at 0.0 useQuery.loading = true
at 0.01 userQuery.loading = false (takes 0.01 to do they query say)
at 1 second userQuery.loading = true
at 1.01 userQuery.loading = false

So what i thought, if button is pressed at 0.3, we want to make it look like it's loading straight away.
Say last poll was at 0.0, and we press button at 0.2, unless we set isLoading on the button straight away it will be pressable  for 0.8, thus the extra isLoading thingy

Reply:
The trouble is that the `syncStatus`  is only updated by the polling interval, which is 1s currently. So the status isn't updated until that interval is hit which could be up to 1s later than when the button is clicked. Ideally the `manualSync` mutation would tell the `syncInfo` query to fetch now, and then we can use the status via react query but I couldn't find a way to do this.

Reply:
> syncInfo query to fetch now

Oh yeah in that case we can listen for loading on syncStatus and manualSync (I didn't want to go too deep into it, but there is also potential that syncStatus is not straight away updated to syncing, since there is a message that is send to another 'worker' when manualSync is pressed). 

Either way I think current logic for toggling isLoading is the simplest

I think even if this happens (useQuery btw should return some method to allow refetch, at least that's what apollo client does)

Reply:
yeah, there is a refetch function available. I used that early on and have moved away from that approach.